### DISCUSSION TITLE
How to use with Nuxt3?

### DISCUSSION THREAD

**Discussion by dosstx** [vuejs/vuefire#1247] [Questions and Help]
I am testing out the possibily of using VueFire@next with Nuxt3 and tried to do the following setup based on the Vuefire docs:

```js
<template>
  <div>
    <ul>
      <li v-for="post in posts" :key="post.id">
        <span>{{ post }}</span>
      </li>
    </ul>
  </div>
</template>

<script setup>
import { useCollection } from 'vuefire'
import { collection } from 'firebase/firestore'

onMounted(() => {
  const { firestore } = useFirebase()
  const posts = useCollection(collection(firestore, 'posts'))
})
</script>
```

The above code causes the following warning in devtools and **nothing is rendered to the screen.**

[Vue warn]: Property "posts" was accessed during render but is not defined on instance. 
  at <Scratch onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > > 
  at <Anonymous key="/scratch" routeProps= {Component: {…}, route: {…}} pageKey="/scratch"  ... > 
  at <BaseTransition mode="out-in" appear=false persisted=false  ... > 
  at <Transition name="page" mode="out-in" > 
  at <RouterView name=undefined route=undefined > 
  at <NuxtPage> 
  at <Default > 
  at <Anonymous key="default" name="default" hasTransition=true > 
  at <BaseTransition mode="out-in" appear=false persisted=false  ... > 
  at <Transition name="layout" mode="out-in" > 
  at <Anonymous> 
  at <App key=1 > 
  at <NuxtRoot>

Note: I am using this code successfully without Vuefire, but I wanted to test out Vuefire as a possible alternative. Thank you!

**Comment by posva** [comment]
There are plans for a nuxt module that will handle the SSR out of the box. At this moment, the SSR is a bit manual to handle so it's better if you don't need it yet.

As for your example, you need to move the posts variable outside of the `onMounted()` or it will simply not be accessible, as the error tells you 😄

**Reply by dosstx** [reply]
Thanks! Yes, I made the following update:

```js
<template>
  <div>
    <ul>
      <li v-for="post in posts" :key="post.id">
        <span>{{ post }}</span>
      </li>
    </ul>
  </div>
</template>

<script setup>
import { useCollection } from 'vuefire'
import { collection } from 'firebase/firestore'

const posts = ref([])

onMounted(() => {
  const { firestore } = useFirebase()
  posts.value = useCollection(collection(firestore, 'posts'))
})
</script>
```

But I have to put the `useFirebase()` composable in the `mounted` hook because otherwise I get the following error:
`Component auth has not been registered yet`

I do now see posts array in devtools but for some reason the `id` property is labeled a `function` ? Thus, using `:key="post.id"` won't work for the UI:

![image](https://user-images.githubusercontent.com/6342102/199053200-13bf481d-4214-4a8c-95bd-03cf63358903.png)

Devtools error:
`Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'id')`

**Reply by posva** [reply]
it's `useCollection(..., { target: posts })` instead

**Reply by dosstx** [reply]
Like this?
```js
posts.value = useCollection(collection(firestore, 'posts'), {
    target: posts
  })
  ```

I tried that now and get this error:
```js
shared.esm-bundler.js:453 
        
       Uncaught (in promise) TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'Object'
    |     property 'deps' -> object with constructor 'Object'
    |     index 0 -> object with constructor 'Object'
    |     property 'Set(1)' -> object with constructor 'Array'
    --- index 0 closes the circle
    at JSON.stringify (<anonymous>)
    at toDisplayString (shared.esm-bundler.js:453:24)
    ```
  

    Sorry to bother....thank you for your time.

**Reply by posva** [reply]
no, without the `posts.value =`

**Comment by posva** [comment]
The module is now documented in VueFire docs!
